src/server/node_func.c

```diff
@@ -242,7 +242,7 @@ char *nodename;
 
 
 static struct pbsnode	*old_address = 0;			/*node in question */
-static short		old_state = (short)0xdead;	/*node's   state   */
+static unsigned long	old_state = (unsigned long)0xdead;	/*node's   state   */
```

Comment:
You should set this to 0xdeadbeef now that it is an unsigned long, or just initialize it to 0x0.